### DISCUSSION TITLE
Alert Rule Hindsights - Allowing testing of rule queries on historical data at design time

### DISCUSSION THREAD

**Discussion by alexweav** [grafana/grafana#53422] [Alerting]
Right now, when designing an alert rule, we can execute the query instantaneously and see whether the rule would be firing or not _right now._ This is useful for designing rules, but adds some uncertainty as you can't be certain when your rule would have fired for past incidents.

We could take this one step further and execute the query on a longer range of historical data, at design time. The "test rule" button would not only show whether the rule is normal/firing right now, it would show you when historically the rule you are designing would have fired.

This drastically improves the UX when creating rules. Users can have confidence that a rule would behave as expected on past incidents, and can have an easier time of tuning their rule thresholds.

**Comment by alexweav** [comment]
Paraphrasing from @gotjosh

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
